onnx/shape_inference/implementation.cc

```diff
@@ -620,17 +659,21 @@ void InferShapeForFunctionNode(
 void InferShapeForFunctionNode(
     const FunctionProto* func,
     const ISchemaRegistry* schema_registry,
-    InferenceContext& ctx) {
+    InferenceContext& ctx,
+    SymbolTableImpl &symbolTable) {
   std::unordered_map<std::string, int> opset_imports;
   for (const auto& opset_import : func->opset_import()) {
     opset_imports[opset_import.domain()] = static_cast<int>(opset_import.version());
   }
-  InferShapeForFunctionNode(func, opset_imports, schema_registry, ctx);
+  InferShapeForFunctionNode(func, opset_imports, schema_registry, ctx, symbolTable);
 }
 
 std::vector<const TypeProto*> GraphInferencerImpl::doInferencing(
     const std::vector<const TypeProto*>& inputTypes,
     const std::vector<const TensorProto*>& inputData) {
+  SymbolTableImpl& symbolTable = getSymbolTable();
+  // add existing symbolic shape into symbol table from sub graph
+  symbolTable.addFromGraph(*g_);
```

Comment:
you dont need this anymore right? since you are already traversing the whole graph including subgraphs in the begining